### PR TITLE
Story: ADAPTEDS-13, Task: ADAPTEDS-49

### PR DESCRIPTION
**Pull request by AmirTalakoob**
### Description
Added pages SignUp and SignUpAdditional, the first one gets email and password, and the second one handles other personal information.
Access restriction is not possible to be defined as of now.

### Solution
Created two JSX files, handling two forms that are designed to (after being connected to the backend) send a JSON containing new user information.
Most of the input validations are done, but \Confirmed password needs to be worked on.

### Testing
Program runs, as well as the regex for the password and other required fields.
The page is connected to the home page, and the buttons function correctly.

### CODE REVIEW

File: frontend/adapted_strength-app/src/pages/SignUp.jsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+*/
+
+
+function EmailField() {
+    return (<input type="email" placeholder="Email Address" id="email" name="email" required/>);
+}
+function PasswordField() {
+    return (<input type="text" placeholder="Password" id="password" name="password" pattern="^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[@$!%*?&])[A-Za-z\d@$!%*?&]{8,255}$" 
+    title="Password must contain at least one lowercase letter, one uppercase letter, one digit, and one special character. It should be between 8 and 255 characters long."
+    /*onchange="validatePasswordMatch()"*/ required/>);
```
Comment (kyperbelt) [inline frontend/adapted_strength-app/src/pages/SignUp.jsx:23]: remove if not needed

File: frontend/adapted_strength-app/src/pages/SignUp.jsx [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+}
+function PasswordConfirmationField() {
+    return (<input type="text" placeholder="Re-Enter Password" id="password_conf" name="password_confirmation" /*onchange="validatePasswordMatch()"*/ required/>);
```
Comment (kyperbelt) [inline frontend/adapted_strength-app/src/pages/SignUp.jsx:26]: remove commented section if not needed

File: frontend/adapted_strength-app/src/pages/SignUp.jsx [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    return (<button type="submit" className="border-slate-50 border-8 bg-black text-slate-200 rounded-full px-3 py-1 "  >Next</button>);
+}
+
+function HandleSubmit(event){
+    event.preventDefault();
+    //if (validatePasswordMatch()){
+        window.location.href="/sign-up-additional";
+        //store info
+        this.reset();
+    //}
```
Comment (kyperbelt) [inline frontend/adapted_strength-app/src/pages/SignUp.jsx:38]: fix indentation and remove comments if not needed

File: frontend/adapted_strength-app/src/pages/SignUp.jsx [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    var password = document.getElementById("password").value;
+    var password_conf = document.getElementById("password_conf").value;
+
+    if (password !== password_conf) {
+        return false;
+    } else {
+        return true;
+    }
+}
+*/
```
Comment (kyperbelt) [inline frontend/adapted_strength-app/src/pages/SignUp.jsx:14]: remove if we dont need

**Review by kyperbelt — CHANGES_REQUESTED** [review]

File: frontend/adapted_strength-app/src/pages/SignUp.jsx (hunk 1 above)
Comment (AmirTalakoob) [reply frontend/adapted_strength-app/src/pages/SignUp.jsx:23]: Mentioned in task: ADAPTEDS-89

File: frontend/adapted_strength-app/src/pages/SignUp.jsx (hunk 2 above)
Comment (AmirTalakoob) [reply frontend/adapted_strength-app/src/pages/SignUp.jsx:26]: Mentioned in task: ADAPTEDS-89

File: frontend/adapted_strength-app/src/pages/SignUp.jsx (hunk 3 above)
Comment (AmirTalakoob) [reply frontend/adapted_strength-app/src/pages/SignUp.jsx:38]: Mentioned in task: ADAPTEDS-89

File: frontend/adapted_strength-app/src/pages/SignUp.jsx (hunk 4 above)
Comment (AmirTalakoob) [reply frontend/adapted_strength-app/src/pages/SignUp.jsx:14]: Mentioned in task: ADAPTEDS-89

**Review by kyperbelt — APPROVED** [review]

**Pull request merged**